// gitbook-plugin-lilypond-highlight/index.js
/* gitbook-plugin-lilypond-highlight
   This plugin only exposes one item: {% lilypond %}, which takes input in the
   GNU LilyPond language and produces syntax highlighting for it.

   There is one external dependency: python-ly
   This means python-ly has to be installed through pip
   (or from https://github.com/wbsoft/python-ly), and the standalone script `ly`
   has to be accessible.
   Depending on the context it *may* be necessary to run Gitbook in a context
   where the PYTHONPATH environment variable has been explicitly set. For example
   this may happen in a Git hook on the server (where no environment is set up
   in a non-login shell).
   */
'use strict';
var ly = require('ly-client');
var _last_prepared = '';

module.exports = {
    book: {
        assets: './assets',
        css: [
            'lilypond.css'
        ]
    },
    blocks: {
        lilypond: {
            process: function(block) {
                if (_last_prepared !== 'lilypond') {
                    ly.prepare('highlight', '', 'full-html=false');
                    _last_prepared = 'lilypond';
                };
                return ly.exec.cli_sync(block.body);
            }
        },
        lilycode: {
            process: function(block) {
                if (_last_prepared !== 'lilycode') {
                    ly.prepare('highlight', '', 'wrapper-tag=code;full-html=false');
                    _last_prepared = 'lilycode';
                };
                return ly.exec.sync(block.body);
            }
        }
    }
};
